src/hooks/query/users/useUpdateUser.js

```diff
@@ -0,0 +1,25 @@
+import { useMutation } from '@tanstack/react-query'
+import { updateUser } from '../../../api'
+import { queryKeys } from '../queryKeys'
+import { useAuthContext } from '../../../contexts/auth'
+
+export const useUpdateUser = () => {
+  const { setLogger, logger } = useAuthContext()
+  const { id } = logger
+
+  const { mutate, isLoading } = useMutation((data) => updateUser(id, { ...data }), {
+    mutationKey: [queryKeys.updateUser],
```

Comment:
better to add additional dependency here: `[queryKeys.updateUser], id`

Plus additional option:

onError: ....., ,
onSuccess: ..... ,
enabled: !!id